source/dofs/dof_tools_constraints.cc

```diff
@@ -1909,7 +1887,7 @@ namespace DoFTools
       // If face_2 does have children, then we need to iterate over these
       // children and set periodic constraints in the inverse direction:
 
-      if (face_2->has_children())
+      if ((!use_mg) && face_2->has_children())
```

Comment:
Same question here: Why does it matter whether we use MG? I guess it's because if you do level constraints, what matters is what happens on *this* level, not whether there are other levels?